tc-root/tc_root_user.c

```diff
@@ -207,10 +207,14 @@ static bool validate_ifname(const char *input_ifname, char *output_ifname) {
     if (len >= IF_NAMESIZE) {
         return false;
     }
+    // check if interface name is null
+    if (*input_ifname == '\0') {
+        return false;
+    }
     for (i = 0; i < len; i++) {
         char c = input_ifname[i];
-
-        if (!(isalpha(c) || isdigit(c)))
+        // Check for each character of input_ifname  if it contains '/' or whitespace characters
+        if (c == '/' || isspace(c))
```

Comment:
I believe both / and space are legal in interface names on windows

Reply:
As discussed in meeting 4/2, change line 71 to change `dev %s` to `dev '%s'`, and also in dev there it needs to escape any ' to \'

Reply:
Okay I will make changes to the line 71.
but what should I do with interface name validation check? Should we bypass it?

Reply:
The call to `if_nametoindex()` in line 278 is the interface name check.  I am not any validation is needed beyond that.
I am not even sure whether line 211 buys us anything since I would expect an empty string to naturally fail line 278.
So why not just delete the entire `validate_ifname()` function, does it really help with anything?

Reply:
Disagree, unless if_nametoindex() crashes, which it should not, I don't see any value in having additional name or length checks.